dashboard/app.js

```diff
@@ -0,0 +1,299 @@
+// app.js - Plain JS dashboard using Chart.js
+
+const COMPLETED_URL = "../completed_commits.json";
+const RESULT_URL = sha => `../results/${sha}/metrics.json`;
+const COMMIT_URL = sha => `https://github.com/valkey-io/valkey/commit/${sha}`;
+
+async function fetchJSON(url){
+  const r = await fetch(url);
+  if(!r.ok) throw new Error(`${url}: ${r.status}`);
+  return r.json();
+}
+
+const state = {
+  commits: [],       // ordered list of shas
+  commitTimes: {},   // sha -> timestamp
+  metrics: [],       // raw metric rows
+  cluster: 'all',
+  tls: 'all',
+  metricKey: 'rps',
+  pipeline: 'all',
+  dataSize: 'all',
+  selectedCommands: new Set(),
+  fromDate: '',
+  toDate: ''
+};
+
+// DOM helpers --------------------------------------------------------------
+function el(tag, attrs={}, children=[]){
+  const e = document.createElement(tag);
+  Object.entries(attrs).forEach(([k,v])=>{
+    if(k.startsWith('on') && typeof v === 'function'){
+      e.addEventListener(k.slice(2).toLowerCase(), v);
+    } else if(k==='class') {
+      e.className = v;
+    } else {
+      e.setAttribute(k,v);
+    }
+  });
+  children.forEach(c=>{ if(typeof c==='string') e.appendChild(document.createTextNode(c));
+                       else if(c) e.appendChild(c); });
+  return e;
+}
+
+function buildControls(){
+  const controls = el('div', {class:'flex flex-wrap gap-4 justify-center text-center mb-4'});
+  const selectCluster = el('select', {class:'border rounded p-1 ml-2',
+    onchange:e=>{state.cluster=e.target.value; updateChart();}},
+    ['all','true','false'].map(o=>el('option',{value:o},[o])));
+  const selectTLS = el('select', {class:'border rounded p-1 ml-2',
+    onchange:e=>{state.tls=e.target.value; updateChart();}},
+    ['all','true','false'].map(o=>el('option',{value:o},[o])));
+  const selectPipeline = el('select', {class:'border rounded p-1 ml-2'});
+  selectPipeline.addEventListener('change', e=>{state.pipeline=e.target.value; updateChart();});
+  const selectSize = el('select', {class:'border rounded p-1 ml-2'});
+  selectSize.addEventListener('change', e=>{state.dataSize=e.target.value; updateChart();});
+  const selectMetric = el('select', {class:'border rounded p-1 ml-2',
+    onchange:e=>{state.metricKey=e.target.value; updateChart();}},
+    ['rps','avg_latency_ms','p95_latency_ms','p99_latency_ms'].map(o=>el('option',{value:o},[o])));
```

Comment:
Regarding latencies. I find p50 a much more important than avg. usually average helps identify anomalies and call out to search investigate the tail latencies. I think in case of benchmark the P50 is more important (can we add it?)